comments/certbot_ssl_for_azure_web_app.md

```diff
@@ -0,0 +1,74 @@
+# Manually generating SSL certs for Azure web apps using Certbot
+
+## Local operations on the CLI
+
+### Generate the cert
+
+Install `certbot`:
+
+    sudo apt install certbot python3-certbot-nginx
+
+Generate a cert for a target domain using DNS challenge type:
+
+    sudo certbot certonly --manual --preferred-challenges dns -d domain.com
+
+This will tell you to add a DNS TXT record for that domain. Most of our domains are now managed through Amazon Route53, so
+go there in your browser, find the Hosted Zone that corresponds to the domain you're generating the cert for, and then add
+the TXT record.
+
+Once the TXT record is saved, tell certbot to continue. It should report success and save your certificate and signing chain to
+`/etc/letsencrypt/live/domain.com`
+
+### Alternatively...
+
+There's a plugin that lets us automate doing the TXT record faffing in Route53. Install with:
+
+    sudo apt install python3-certbot-dns-route53
+
+This uses standard AWS CLI credentials, but since certbot needs to be run as root, we need to symlink our local user's AWS
+credentials directory to an equivalent location for root:
+
+    sudo -s
+    ln -s /home/alightwing/.aws ~/.aws
+    exit
+
+Then if we want to renew the cert, all we need to do is:
+
+    sudo certbot certonly --dns-route53 -d domain.com
+
+This won't renew the cert if it's not close to expiring. To force a renewal:
+
+    sudo certbot certonly --dns-route53 -d domain.com --force-renewal
+
+### Convert Let's Encrypt cert files to Private Certificate
+
+Azure cert attachments need to come in the form of Private Key Certificates (.pfx) so we need to convert our .pem files from 
+Let's Encrypt to .pfx. This is done with `openssl`:
+
+    sudo openssl pkcs12 \
```

Comment:
I also had to `chown` this after, as the converted cert is owned by root, and my browser didn't have permission to read it.